# 1. Business Context

A retail company has accumulated customer demographic, purchasing, and campaign response data over several years. While this information is available, marketing decisions are currently broad and largely untargeted, resulting in inefficient promotional spending and inconsistent customer engagement.

The company wants to better understand its customers by identifying meaningful customer groups based on purchasing behavior, demographics, and campaign interactions. These insights will support more personalized marketing strategies, stronger customer retention, and improved allocation of marketing resources.

Rather than treating every customer the same, the objective is to transform historical customer data into actionable business intelligence that supports strategic decision-making.

# 2. Problem Satetment
The current marketing strategy treats customers similarly despite significant differences in purchasing behavior and customer value. Without a structured understanding of customer segments, marketing campaigns become less effective, customer engagement decreases, and opportunities for revenue growth remain untapped.

# 3. Project Objectives

## Primary Objective

Develop a customer intelligence platform that transforms customer transaction and demographic data into actionable marketing insights.

## Supporting Objectives

* Understand customer demographics and purchasing behavior.
* Identify meaningful customer segments.
* Build business-friendly customer personas.
* Recommend marketing actions for each customer group.
* Simulate potential revenue improvements from targeted campaigns.
* Present findings through an interactive analytics dashboard.

# 4. Business Questions

To achieve the project objectives, the following business questions will guide the analysis throughout the project.

## Customer Understanding

- Who are the company's most valuable customers based on purchasing behaviour?
- Which customers purchase most frequently and generate the highest revenue?
- How do demographic characteristics such as age, income, education, and marital status influence purchasing behaviour?
- Are there noticeable differences in spending habits across customer groups?

---

## Marketing Strategy

- Which customers respond most positively to previous marketing campaigns?
- Which customer groups should receive personalized promotions?
- Which purchasing channels (Web, Store, or Catalog) are preferred by different customer groups?
- Can campaign performance be improved through customer segmentation?

---

## Business Decision Making

- Can customers be grouped into meaningful segments based on their behavioural and demographic characteristics?
- Which customer segment contributes the highest business value?
- Which customer segments should be prioritized for loyalty programs and retention strategies?
- What potential business impact could personalized marketing campaigns generate?

# 5. Dataset Overview

The dataset contains historical customer information collected by a retail company. It includes demographic details, purchasing behaviour, campaign responses, and customer engagement metrics.

This section provides an initial overview of the dataset by loading the data, inspecting its structure, and reviewing the available features before performing any cleaning or preprocessing.

In [12]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

In [13]:
df = pd.read_csv("../data/raw/marketing_campaign.csv", sep="\t")

In [14]:
df.head()

,ID,Year_Birth,Education,Marital_Status,Income,Kidhome,Teenhome,Dt_Customer,Recency,MntWines,MntFruits,MntMeatProducts,MntFishProducts,MntSweetProducts,MntGoldProds,NumDealsPurchases,NumWebPurchases,NumCatalogPurchases,NumStorePurchases,NumWebVisitsMonth,AcceptedCmp3,AcceptedCmp4,AcceptedCmp5,AcceptedCmp1,AcceptedCmp2,Complain,Z_CostContact,Z_Revenue,Response
0,5524,1957,Graduation,Single,58138.0,0,0,04-09-2012,58,635,88,546,172,88,88,3,8,10,4,7,0,0,0,0,0,0,3,11,1
1,2174,1954,Graduation,Single,46344.0,1,1,08-03-2014,38,11,1,6,2,1,6,2,1,1,2,5,0,0,0,0,0,0,3,11,0
2,4141,1965,Graduation,Together,71613.0,0,0,21-08-2013,26,426,49,127,111,21,42,1,8,2,10,4,0,0,0,0,0,0,3,11,0
3,6182,1984,Graduation,Together,26646.0,1,0,10-02-2014,26,11,4,20,10,3,5,2,2,0,4,6,0,0,0,0,0,0,3,11,0
4,5324,1981,PhD,Married,58293.0,1,0,19-01-2014,94,173,43,118,46,27,15,5,5,3,6,5,0,0,0,0,0,0,3,11,0


In [15]:
df.shape

(2240, 29)

In [16]:
df.columns

Index(['ID', 'Year_Birth', 'Education', 'Marital_Status', 'Income', 'Kidhome',
       'Teenhome', 'Dt_Customer', 'Recency', 'MntWines', 'MntFruits',
       'MntMeatProducts', 'MntFishProducts', 'MntSweetProducts',
       'MntGoldProds', 'NumDealsPurchases', 'NumWebPurchases',
       'NumCatalogPurchases', 'NumStorePurchases', 'NumWebVisitsMonth',
       'AcceptedCmp3', 'AcceptedCmp4', 'AcceptedCmp5', 'AcceptedCmp1',
       'AcceptedCmp2', 'Complain', 'Z_CostContact', 'Z_Revenue', 'Response'],
      dtype='str')

In [17]:
df.dtypes

ID                       int64
Year_Birth               int64
Education                  str
Marital_Status             str
Income                 float64
Kidhome                  int64
Teenhome                 int64
Dt_Customer                str
Recency                  int64
MntWines                 int64
MntFruits                int64
MntMeatProducts          int64
MntFishProducts          int64
MntSweetProducts         int64
MntGoldProds             int64
NumDealsPurchases        int64
NumWebPurchases          int64
NumCatalogPurchases      int64
NumStorePurchases        int64
NumWebVisitsMonth        int64
AcceptedCmp3             int64
AcceptedCmp4             int64
AcceptedCmp5             int64
AcceptedCmp1             int64
AcceptedCmp2             int64
Complain                 int64
Z_CostContact            int64
Z_Revenue                int64
Response                 int64
dtype: object

In [18]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 29 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2240 non-null   int64  
 1   Year_Birth           2240 non-null   int64  
 2   Education            2240 non-null   str    
 3   Marital_Status       2240 non-null   str    
 4   Income               2216 non-null   float64
 5   Kidhome              2240 non-null   int64  
 6   Teenhome             2240 non-null   int64  
 7   Dt_Customer          2240 non-null   str    
 8   Recency              2240 non-null   int64  
 9   MntWines             2240 non-null   int64  
 10  MntFruits            2240 non-null   int64  
 11  MntMeatProducts      2240 non-null   int64  
 12  MntFishProducts      2240 non-null   int64  
 13  MntSweetProducts     2240 non-null   int64  
 14  MntGoldProds         2240 non-null   int64  
 15  NumDealsPurchases    2240 non-null   int64  
 16 

In [19]:
df[["Z_CostContact", "Z_Revenue"]].head()

,Z_CostContact,Z_Revenue
0,3,11
1,3,11
2,3,11
3,3,11
4,3,11


In [20]:
df["Z_CostContact"].value_counts()

Z_CostContact
3    2240
Name: count, dtype: int64

In [21]:
df["Z_Revenue"].value_counts()

Z_Revenue
11    2240
Name: count, dtype: int64

# 6. Feature Categorization (Data Dictionary)

To better understand the dataset, the available features are grouped into business-oriented categories rather than being interpreted individually.

This categorization improves readability and provides a clear understanding of the information available for customer analysis, feature engineering, and customer segmentation in the later stages of the project.

The dataset consists of the following feature groups:

- Customer Demographics
- Customer Relationship
- Product Spending
- Purchase Behaviour
- Marketing Campaign Interaction
- Administrative Variables





## 6.1 Customer Demographics

These features describe the customer's personal and socio-economic characteristics.

| Feature | Description |
|----------|-------------|
| **ID** | Unique customer identifier |
| **Year_Birth** | Customer's birth year |
| **Education** | Highest education level |
| **Marital_Status** | Customer's marital status |
| **Income** | Annual household income |
| **Kidhome** | Number of children living at home |
| **Teenhome** | Number of teenagers living at home |

## 6.2 Customer Relationship

These features describe the customer's relationship with the company.

| Feature | Description |
|----------|-------------|
| **Dt_Customer** | Date the customer enrolled with the company |
| **Recency** | Number of days since the customer's most recent purchase |

## 6.3 Product Spending

These features represent the amount spent by each customer across different product categories.

| Feature | Description |
|----------|-------------|
| **MntWines** | Amount spent on wine products |
| **MntFruits** | Amount spent on fruits |
| **MntMeatProducts** | Amount spent on meat products |
| **MntFishProducts** | Amount spent on fish products |
| **MntSweetProducts** | Amount spent on sweet products |
| **MntGoldProds** | Amount spent on gold products |

## 6.4 Purchase Behaviour

These features capture how customers interact with different purchasing channels.

| Feature | Description |
|----------|-------------|
| **NumDealsPurchases** | Purchases made using discounts or special deals |
| **NumWebPurchases** | Purchases made through the website |
| **NumCatalogPurchases** | Purchases made through the product catalog |
| **NumStorePurchases** | Purchases made in physical stores |
| **NumWebVisitsMonth** | Number of website visits in the last month |

## 6.5 Marketing Campaign Interaction

These variables measure customer engagement with previous marketing campaigns.

| Feature | Description |
|----------|-------------|
| **AcceptedCmp1** | Accepted Campaign 1 |
| **AcceptedCmp2** | Accepted Campaign 2 |
| **AcceptedCmp3** | Accepted Campaign 3 |
| **AcceptedCmp4** | Accepted Campaign 4 |
| **AcceptedCmp5** | Accepted Campaign 5 |
| **Complain** | Indicates whether the customer has submitted a complaint |
| **Response** | Indicates whether the customer accepted the most recent campaign |

## 6.6 Administrative Variables

These variables appear to be administrative fields originating from the company's internal database.

During the initial inspection, both variables were found to contain only a single unique value across all customer records. Since they exhibit zero variance, they are unlikely to contribute meaningful information for customer segmentation or downstream analysis and will be evaluated further during the data preprocessing stage.

| Feature | Description |
|----------|-------------|
| **Z_CostContact** | Administrative constant field (requires validation) |
| **Z_Revenue** | Administrative constant field (requires validation) |

# 7. Key Observations

Based on the initial exploration of the dataset, the following observations have been made:

- The dataset contains **2,240 customer records** and **29 features**, providing sufficient information for customer behaviour analysis and segmentation.
- The available features cover multiple business dimensions, including customer demographics, purchasing behaviour, campaign interactions, and customer relationship history.
- The majority of the variables are numerical, making the dataset well suited for statistical analysis and clustering techniques after appropriate preprocessing.
- The **Income** feature contains a small number of missing values and will require further investigation during the data quality assessment stage.
- The **Dt_Customer** feature is currently stored as a text field and will later be converted into a datetime format to derive additional business features such as customer tenure.
- The administrative variables **Z_CostContact** and **Z_Revenue** appear to contain constant values across all customers and are likely to provide little analytical value. Their usefulness will be validated during preprocessing.
- Overall, the dataset appears to be comprehensive enough to support customer intelligence, segmentation, and marketing analytics objectives.